evm/src/cpu/kernel/interpreter.rs

```diff
@@ -56,10 +56,12 @@ impl MemoryState {
     }
 }
 
-pub(crate) struct Interpreter<'a> {
+pub(crate) struct Interpreter<'a, F: Field> {
```

Comment:
This is what causes all the noise in this PR yes? Do you think having `new_with_state_and_halt_condition()` be a freestanding function instead could work so you don't have to add the generic here? So you'd have `fn new_interpreter_with_state<F: Field>(… … )` outside the `impl` for `Interpreter`?

Reply:
But you would need to create a new `Intepreter<F>` within `new_interpreteter_with_state`. The problem is that right now `Interpreter` has a field representing the state of type `GenerationState<GoldilocksField>`, but we would like to run it in a context where the state is only `GenerationState<F>` for `F: Field`.